src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c

```diff
@@ -4716,14 +4731,21 @@ static ssize_t
 kds_cucounts_show(struct device *dev, struct device_attribute *attr, char *buf)
 {
 	struct exec_core *exec = dev_get_exec(dev);
+	struct xocl_ert *xert = exec_is_ert(exec) ? exec->ert : NULL;
 	unsigned int cus = exec ? exec->num_cus - exec->num_cdma : 0;
 	unsigned int sz = 0;
 	unsigned int idx;
 
 	for (idx = 0; idx < cus; ++idx) {
 		struct xocl_cu *xcu = exec->cus[idx];
 
-		sz += sprintf(buf, "cu[%d] done(%d) run(%d)\n", idx, xcu->done_cnt, xcu->run_cnt);
+		sz += sprintf(buf + sz, "cu[%d] done(%d) run(%d)\n", idx, xcu->done_cnt, xcu->run_cnt);
+	}
+
+	cus = exec ? exec->num_sk_cus : 0;
+	for (idx = exec->num_cus; idx < exec->num_cus + cus; idx++) {
+		sz += sprintf(buf + sz, "cu[%d] done(%d) run(%d)\n", idx,
+			ert_cu_usage(xert, idx), ert_cu_usage(xert, idx));
```

Comment:
Potential code crash here.   There is a possibility that xert is a null pointer (line 4734) and `ert_cu_usage() `doesn't check for null pointers.  Same goes for `ert_cu_usage()`.

Note Line 4782 does get around this issue by first checking for NULL and then calling the appropriate function.

Reply:
With soft kernel (exec->num_sk_cus > 0), ert should not be null. panic would be ok here because that means a bug.